src/ResourceManagement/Insights/Insights/Customizations/Shoebox/SasMetricRetriever.cs

```diff
@@ -23,18 +23,24 @@ internal abstract class SasMetricRetriever : IMetricRetriever
         {
             MetricFilter filter = MetricFilterExpressionParser.Parse(filterString);
 
+            // Filter definitions by timegrain
+            var timegraindefinitions = from d in definitions
+                                       where d.MetricAvailabilities.Count > 0
+                                       && d.MetricAvailabilities[0].TimeGrain == filter.TimeGrain
```

Comment:
This will not work for metrics with multiple allowed timegrains, for which the query specifies one other than the first. It needs to be able to select the appropriate availability based on the requested timegrain.